src/utils/encoding.ts

```diff
@@ -171,3 +173,45 @@ export function parseReplaceRequest(req: ReplaceRequest, network: Network): Repl
         status: req.status,
     };
 }
+
+export function parseIssueRequest(req: IssueRequest, network: Network, id: H256 | string): Issue {
+    const status = req.status.isCompleted
+        ? IssueStatus.Completed
+        : req.status.isCancelled
+            ? IssueStatus.Cancelled
+            : IssueStatus.PendingWithBtcTxNotFound;
+    return {
+        id: stripHexPrefix(id.toString()),
+        creationBlock: req.opentime.toNumber(),
+        vaultBTCAddress: encodeBtcAddress(req.btc_address, network),
+        vaultDOTAddress: req.vault.toString(),
+        userDOTAddress: req.requester.toString(),
+        vaultWalletPubkey: req.btc_public_key.toString(),
+        bridgeFee: BTCAmount.from.Satoshi(req.fee.toString()),
+        amountInterBTC: BTCAmount.from.Satoshi(req.amount.toString()),
+        griefingCollateral: PolkadotAmount.from.Planck(req.griefing_collateral.toString()),
+        status,
+    };
+}
+
+export function parseRedeemRequest(req: RedeemRequest, network: Network, id: H256): Redeem {
+    const status = req.status.isCompleted
+        ? RedeemStatus.Completed
+        : req.status.isRetried
+            ? RedeemStatus.Retried
```

Comment:
Huh, I tried to use prettier to change the formatting but then eslint complained. Maybe we should look into fixing the linting configs so they can work together (later, of course).

Reply:
I completely stopped using prettier. To format, I now run `yarn fix` which applies eslint at the end.